### PR TITLE
Added headerbar

### PR DESCRIPTION
**Pull request by phntxx**
Added headerbar as fix for issue #9

### CODE REVIEW

File: frontend/src/config.ts [hunk 1]
```diff
@@ -60,3 +60,8 @@ export const mediaBreakpoints = {
   phone: "767px",
   tablet: "1024px",
 };
+
+/**
+ * The height of the top navigation bar.
+ */
+export const headerBarHeight = "50px";
```
Comment (michaelneu) [inline frontend/src/config.ts]: Setting the height in a fixed pixel size is doomed to cause trouble. Rather have padding in `rem`, which grows with the content.

File: frontend/src/components/dashboard.tsx [hunk 2]
```diff
@@ -12,7 +12,10 @@ import { Settings } from "./settings";
 import { ConnectedSidebar } from "./sidebar";
 import { SidebarBurger } from "./sidebar-burger";
 
-interface ISidebarAwareProps {
+/**
+ * Interface for whether or not the sidebar is being shown.
+ */
+export interface ISidebarAwareProps {
```
Comment (michaelneu) [inline frontend/src/components/dashboard.tsx]: I'm not a fan of sharing props definitions across different modules, but it should be fine here.

Comment (phntxx) [reply frontend/src/components/dashboard.tsx]: Only did it that way so that I could effortlessly move the headerbar into a seperate file, as I think putting it into the same file would just clutter the code / the file.

File: frontend/src/components/headerbar.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  top: 0px;
+  left: 0px;
+  width: 100%;
+  height: ${ headerBarHeight };
+
+  z-index: 1;
+
+  background-color: white;
+  box-shadow: 0 4px 2px -2px gray;
```
Comment (michaelneu) [inline frontend/src/components/headerbar.tsx]: `gray` is too hard, use something like `rgba(0, 0, 0, 0.1)`

File: frontend/src/components/headerbar.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Header bar for all views. Usually shows control to show/hide sidebar.
+ */
+export const HeaderBar = styled.div<ISidebarAwareProps>`
+  position: fixed;
+  display: flex;
+
+  top: 0px;
+  left: 0px;
+  width: 100%;
```
Comment (michaelneu) [inline frontend/src/components/headerbar.tsx]: Rather fix it to left and right, instead of having `100%` width. This can cause issues when the sidebar is open, as the width stays unchanged, but the container moves to the right.

File: frontend/src/components/users.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+import { Heading } from "./headings";
+
+/**
+ * A page to list all of the users that have registered.
+ */
+export const Users = () => {
```
Comment (michaelneu) [inline frontend/src/components/users.tsx]: `Users` shouldn't be part of this PR, since it's a separate page not affecting the header bar.

File: frontend/src/components/statistics.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
+import { Heading } from "./headings";
+
+/**
+ * A statistics page. Should show things like number of attendees, T-Shirt sizes etc.
+ */
+export const Statistics = () => {
```
Comment (michaelneu) [inline frontend/src/components/statistics.tsx]: Similarly, `Statistics` shouldn't be in this PR.

File: frontend/src/index.html [hunk 7]
```diff
@@ -1,6 +1,7 @@
 <!doctype html>
 <html>
   <head>
+    <link rel="stylesheet" href="node_modules/react-table/react-table.css">
```
Comment (michaelneu) [inline frontend/src/index.html]: This won't work, as webpack can't resolve the `link` to include `react-table` in the bundle. Rather import it in `index.tsx` when necessary.

File: frontend/package.json [hunk 8]
```diff
@@ -40,7 +40,7 @@
     "monaco-editor-webpack-plugin": "^1.7.0",
     "node-uuid": "^1.4.8",
     "react": "^16.8.6",
-    "react-dom": "^16.8.6",
+    "react-dom": "^16.10.0",
```
Comment (michaelneu) [inline frontend/package.json]: Why is it required for the header bar to upgrade `react-dom`?

File: frontend/package.json [hunk 9]
```diff
@@ -62,5 +62,9 @@
     "webpack": "^4.40.2",
     "webpack-cli": "^3.3.0",
     "webpack-dev-server": "^3.8.1"
+  },
+  "dependencies": {
+    "@types/react-table": "^6.8.5",
+    "react-table": "^6.10.3"
```
Comment (michaelneu) [inline frontend/package.json]: `react-table` is not used except for the CSS file. Please don't add dependencies when they're not needed yet. Also, they should be in `devDependencies`, since we only have a build step for the frontend, no runtime.

File: frontend/src/index.html (hunk 7 above)
Comment (phntxx) [reply frontend/src/index.html]: It did, but still get what you're saying
